docs/docs/dev_docs/contracts/syntax/storage/storage_slots.md

```diff
@@ -0,0 +1,64 @@
+In the end a siloed note hash is computed in the kernel. 
+
+:::info
+Some of the syntax below is a little butchered to make it easier to follow variables without the full code.
+:::
+
+```mermaid
+sequenceDiagram
+    alt Call
+    Token->>BalanceMap: Map::new(map_slot);
+    Token->>Token: to_bal = storage.balances.at(to)
+    Token->>BalanceMap: BalanceMap.at(to)
+    BalanceMap->>BalanceMap: derived_slot = H(map_slot, to)
+    BalanceMap->>BalanceSet: BalanceSet::new(to, derived_slot)
+    Token->>BalanceSet: to_bal.add(amount)
+    BalanceSet->>BalanceSet: note = TokenNote::new(amount, to)
+    BalanceSet->>Set: insert(note)
+    Set->>LifeCycle: create_note(derived_slot, note)
+    LifeCycle->>LifeCycle: note.header = NoteHeader { contract_address, <br> storage_slot: derived_slot, nonce: 0, is_transient: true }
+    LifeCycle->>Utils: compute_inner_note_hash(note)
+    Utils->>TokenNote: compute_note_hash(note)
+    TokenNote->>Utils: note_hash = H(amount, to, randomness)
+    Utils->>NoteHash: compute_inner_hash(derived_slot, note_hash)
+    NoteHash->>LifeCycle: inner_note_hash = H(derived_slot, note_hash)
+    LifeCycle->>Context: push_new_note_hash(inner_note_hash)
+    end
+    Context->>Kernel: siloed_note_hash = H(contract_address, inner_note_hash)
+```
+
+To clarify what the commitment inserted from above (the `siloed_note_hash`) really is we "unroll" the values to their simplest components. This gives us a better idea around what is actually inserted into the tree.
+
+```rust
+siloed_note_hash = H(contract_address, inner_note_hash)
+siloed_note_hash = H(contract_address, H(derived_slot, note_hash))
+siloed_note_hash = H(contract_address, H(H(map_slot, to), note_hash))
+siloed_note_hash = H(contract_address, H(H(map_slot, to), H(amount, to, randomness)))
+```
+
+Where the `map_slot` is the slot specified in `Storage::init`, recall:
+
+#include_code storage_balances_init yarn-project/noir-contracts/src/contracts/token_contract/src/main.nr rust
+
+And `to` is the actor who receives the note, `amount` of the note and `randomness` is the randomness used to make the note hiding. Without the `randomness` the note could could just as well be plaintext (computational cost of a preimage attack would be trivial in such a case).
+
+:::info
+Beware that this hash computation is what the aztec.nr libraries is doing, and not strict requirements by the network (only the kernel computation is).
+:::
+
+With this note structure, the contract can require that only notes sitting at specific storage slots can be used by specific operations, e.g., if transferring funds from `from` to `to`, the notes to destroy should be linked to `H(map_slot, from)` and the new notes (except the change-note) should be linked to `H(map_slot, to)`.
+
+That way, we can have logical storage slots, without them really exiting. This means that knowing the storage slot for a note is not enough to actually figure out what is in there (which it is in public).
```

Comment:
```suggestion
That way, we can have logical storage slots, without them really existing. This means that knowing the storage slot for a note is not enough to actually figure out what is in there (whereas it would be for looking up public state).
```